lib/fluent/output.rb

```diff
@@ -557,16 +557,20 @@ def emit(tag, es, chain)
       @emit_count += 1
       formatted_data = {}
       es.each {|time,record|
-        tc = time / @time_slice_cache_interval
-        if @before_tc == tc
-          key = @before_key
-        else
-          @before_tc = tc
-          key = @time_slicer.call(time)
-          @before_key = key
+        begin
+          tc = time / @time_slice_cache_interval
+          if @before_tc == tc
+            key = @before_key
+          else
+            @before_tc = tc
+            key = @time_slicer.call(time)
+            @before_key = key
+          end
+          formatted_data[key] ||= ''
+          formatted_data[key] << format(tag, time, record)
```

Comment:
I see. a) is better because effect is minimum. I will change it.
v0.14 with new API, we should re-consider error stream use.